templates/distribution/scripts/pre-apply.sh.tpl

```diff
@@ -50,4 +50,45 @@ deleteLoki
 
 {{- end }} # end distributionModulesLoggingType
 
+{{- if index .reducers "distributionModulesPolicyType" }}
+
+deleteGatekeeper() {
+
```

Comment:
maybe we could start by deleting gatekeeper's admission and mutating webhooks here, to avoid potential issues while deleting gatekeeper resources.

Reply:
will do

Reply:
I also added the deletion of the validating webhook generated by kyverno, that are leftover in case of deletion (sometimes kyverno clean them up sometimes it will not)